x/wasm/module_adapter.go

```diff
@@ -160,6 +167,32 @@ func (am AppModule) UpgradeHeight() int64 {
 	return types2.GetVenus2Height()
 }
 
+var (
+	once        sync.Once
+	gWasmConfig types.WasmConfig
+	gWasmDir    string
+)
+
+func WasmDir() string {
+	once.Do(Init)
+	return gWasmDir
+}
+
+func WasmConfig() types.WasmConfig {
+	once.Do(Init)
+	return gWasmConfig
+}
+
+func Init() {
```

Comment:
Suggest using init instead of Init and use onec.do in init.  Thus WasmDir and WasmConfig can return gWasmConfig or gWasmDir directly

Reply:
> Suggest using init instead of Init and use onec.do in init. Thus WasmDir and WasmConfig can return gWasmConfig or gWasmDir directly

Had tried it and found that viper config was null because viper was not initialized in `init()`.